cargo-shuttle/src/init.rs

```diff
@@ -50,963 +44,178 @@ impl Template {
-            let route = warp::any().map(|| "Hello, World!");
-            Ok(route.boxed().into())
-        }"#}
+    fn get_sub_path(&self) -> Option<&'static str> {
+        Some("warp/hello-world")
     }
 }
 
 pub struct ShuttleInitThruster;
 
 impl ShuttleInit for ShuttleInitThruster {
-    fn set_cargo_dependencies(
-        &self,
-        dependencies: &mut Table,
-        manifest_path: &Path,
-        url: &Url,
-        get_dependency_version_fn: GetDependencyVersionFn,
-    ) {
-        set_key_value_dependency_version(
-            "shuttle-thruster",
-            dependencies,
-            manifest_path,
-            url,
-            true,
-            get_dependency_version_fn,
-        );
-
-        set_inline_table_dependency_version(
-            "thruster",
-            dependencies,
-            manifest_path,
-            url,
-            false,
-            get_dependency_version_fn,
-        );
-
-        set_inline_table_dependency_features(
-            "thruster",
-            dependencies,
-            vec!["hyper_server".to_string()],
-        );
-
-        set_key_value_dependency_version(
-            "tokio",
-            dependencies,
-            manifest_path,
-            url,
-            true,
-            get_dependency_version_fn,
-        );
+    fn get_repo_url(&self) -> &'static str {
+        "http://github.com/shuttle-hq/shuttle-examples.git"
     }
 
-    fn get_boilerplate_code_for_framework(&self) -> &'static str {
-        indoc! {r#"
-        use thruster::{
-            context::basic_hyper_context::{generate_context, BasicHyperContext as Ctx, HyperRequest},
-            m, middleware_fn, App, HyperServer, MiddlewareNext, MiddlewareResult, ThrusterServer,
-        };
-        
-        #[middleware_fn]
-        async fn hello(mut context: Ctx, _next: MiddlewareNext<Ctx>) -> MiddlewareResult<Ctx> {
-            context.body("Hello, World!");
-            Ok(context)
-        }
-        
-        #[shuttle_runtime::main]
-        async fn thruster() -> shuttle_thruster::ShuttleThruster<HyperServer<Ctx, ()>> {
-            let server = HyperServer::new(
-                App::<HyperRequest, Ctx, ()>::create(generate_context, ()).get("/", m![hello]),
-            );
-            
-            Ok(server.into())
-        }"#}
+    fn get_sub_path(&self) -> Option<&'static str> {
+        Some("thruster/hello-world")
     }
 }
 
 pub struct ShuttleInitNoOp;
 impl ShuttleInit for ShuttleInitNoOp {
-    fn set_cargo_dependencies(
-        &self,
-        _dependencies: &mut Table,
-        _manifest_path: &Path,
-        _url: &Url,
-        _get_dependency_version_fn: GetDependencyVersionFn,
-    ) {
+    fn get_repo_url(&self) -> &'static str {
+        "http://github.com/shuttle-hq/shuttle-examples.git"
     }
 
-    fn get_boilerplate_code_for_framework(&self) -> &'static str {
-        ""
+    fn get_sub_path(&self) -> Option<&'static str> {
+        todo!()
```

Comment:
I'll need to create an empty project for this in `shuttle-examples`

Reply:
Please see https://github.com/shuttle-hq/shuttle-examples/pull/57

Reply:
I can fix a better (useful) version of https://github.com/shuttle-hq/shuttle-examples/pull/57 so that this is resolved.

Reply:
https://github.com/shuttle-hq/shuttle-examples/pull/59